extensions/wikia/SEOTweaks/SEOTweaksHooksHelper.class.php

```diff
@@ -361,4 +361,40 @@ public static function onLinkEnd( $skin, Title $target, array $options, &$text,
 
 		return true;
 	}
+
+	protected static function findLanguagePath( array $parsedUrl ): string {
+		if ( !array_key_exists( 'path', $parsedUrl ) ) {
+			return '';
+		}
+
+		$path = $parsedUrl['path'];
+		if ( strlen( $path ) === 0 ) {
+			return '';
+		}
+
+		$langCode = explode( '/', $path )[1];
+		$languages = Language::getLanguageNames();
+		if ( isset( $languages[$langCode] ) ) {
+			return '/' . $langCode;
+		}
+
+		return '';
+	}
+
+	public static function onLinkerMakeExternalLink( string &$url, string &$text, bool &$link, array &$attribs ): bool {
+		$parsed = parse_url( $url );
+
+		if ( $parsed !== false ) {
+			$host = $parsed['host'];
+			$path = self::findLanguagePath( $parsed );
+			$city_id = WikiFactory::DomainToID( wfNormalizeHost( $host . $path ) );
+			if ( $city_id ) {
+				$primaryCityUrl = parse_url( WikiFactory::cityIDtoDomain( $city_id ) );
```

Comment:
`cityIDtoDomain` ignores the language path, you should use the whole info returned by `cityIDtoUrl`